modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/planner/JoinCommutePlannerTest.java

```diff
@@ -76,6 +82,100 @@ public class JoinCommutePlannerTest extends AbstractPlannerTest {
         );
     }
 
+    /** Ensures that join commute rules are disabled if joins count is too high. Checks the tables order in the query. */
+    @Test
+    public void testCommuteDisabledForManyJoins() throws Exception {
```

Comment:
Test will fail after https://issues.apache.org/jira/browse/IGNITE-24624

Reply:
Probably. I'll fix it in the next ticket. We can use disable_rule or change join types. Looks simpl.